### ISSUE TITLE
where to put the module and submodule meta-includes

### ISSUE THREAD

**Issue by h-2** [seqan/seqan3#54]
Currently we have
```
alphabet.hpp
alphabet/...
```

Instead we could:
```
alphabet/all.hpp
alphabet/...
```

~~The second is the model used by range-v3~~ and it has some benefits which is why I would suggest switching to it:
* better encapsulation (**every header** that belongs to the (sub)module is **in** the folder), this also makes doxygen easier to understand because folders are identical with (sub)modules
* clear distinction between "meta-headers" and real headers, e.g. by just looking at the files:
  ```
  alphabet/concept.hpp
  alphabet/nucleotide.hpp
  ```
  you don't know whether `nucleotide.hpp` is a meta-header or a real header, without also going through the list of folders.

**Comment by h-2** [comment]
@seqan/all what's your opinion?

**Comment by joergi-w** [comment]
I agree with the second version, because I see the benefits you stated. Are there any drawbacks?

Statements like `#include <seqan3/alphabet/*>` would be the nicest way, but I assume that style is not possible...

**Comment by h-2** [comment]
@kneubert Could you do this? It just means moving every meta-header into it's folder and renaming it to `all.hpp`, as well as updating anything that includes it...

**Comment by h-2** [comment]
@kneubert Is there any progress here? Do you have any questions regarding the issue?

**Comment by kneubert** [comment]
@h-2 For me it looks like range-v3 is handling the meta-headers like we have it now: They are outside of the submodule folders and there is no `all.hpp`, for example in `numeric.hpp`. I wonder if there is a convention. 
range-v3/include/range/v3/numeric.hpp
```
#include <range/v3/numeric/accumulate.hpp>
#include <range/v3/numeric/adjacent_difference.hpp>
#include <range/v3/numeric/iota.hpp>
...
```
The `aminoacid` submodule seems to already have the desired structure with `all.hpp` inside.

**Comment by h-2** [comment]
@kneubert You are right, I misunderstood the "all.hpp" headers in the range library. Maybe we should discuss this again? It seems like range-v3, boost and most other libs have our old model (the first one above)....
@rrahn what do you think?

**Comment by h-2** [comment]
@kneubert I discussed this again with @rrahn and we think we should still go with the new model! And you are right, the aminoacid submodule is already in the new style!

**Comment by h-2** [comment]
@kneubert any progress here?

**Comment by h-2** [comment]
fixed by #90
